node-zerox/src/index.ts

```diff
@@ -35,6 +39,12 @@ export const zerox = async ({
   const aggregatedMarkdown: string[] = [];
   const startTime = new Date();
 
+  if (maxTesseractWorkers !== -1 && maxTesseractWorkers < NUM_STARTING_WORKERS) {
```

Comment:
Let's move this to below the validation. No reason to start up workers if one of the inputs is missing

Reply:
Or maybe to `convertPdfToImages` since that is where it is getting used.

Reply:
I'll move it below the validation here. 

Here is a bit more detail on the workings and the idea behind this:

The worker creation is broken up into two places. Here, the goal is to have X number of workers set up by default so that when it's needed, these workers are ready to go.

I'm not using await here or in the other place because if we wait, then we'll lose any gains we made. Instead, this code will be running in parallel, and there are tasks in between that'll take sufficiently long for it to complete (for eg. pdf download). 

If the PDF is larger and requires more workers, then we add those in `convertPdfToImages`, though all of the workers might not have been added by the time the Tesseract is called. Thats still not that big of a worry because as soon as a worker is added to the scheduler, it is assigned a job. So, using it earlier just makes sure that we hit the ground running, while saving time by not "awaiting" and if it needs more workers, we'll add those.

Does this make sense?